### PR TITLE
Engine.Run() should return the full list of performed measurements (fixes #2187)

### LINKED ISSUE #2187: Reports for InProcess jobs don't include non-Result measurements

### PR DESCRIPTION
**Pull request by AndreyAkinshin**

### CODE REVIEW

File: src/BenchmarkDotNet/Engines/Engine.cs [hunk 1]
```diff
@@ -41,6 +41,7 @@ public class Engine : IEngine
         private bool EvaluateOverhead { get; }
         private bool MemoryRandomization { get; }
 
+        private readonly List<Measurement> jittingMeasurements = new (10);
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Engines/Engine.cs:44]: Is my understanding correct that we won't ever have more than 4 jitting measurements?

- overhead jitting, unroll factor 16
- overhead jitting, unroll factor 1
- workload jitting, unroll factor 16
- workload jitting, unroll factor 1

```suggestion
        private readonly List<Measurement> jittingMeasurements = new ();
```

**Review by adamsitnik — APPROVED** [review]
LGTM, but please consider re-adding the removed public properties. Thank you @AndreyAkinshin !

File: tests/BenchmarkDotNet.IntegrationTests/EngineTests.cs [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            Output.WriteLine("*** AllMeasurements ***");
+            foreach (var measurement in measurements)
+                Output.WriteLine(measurement.ToString());
+            Output.WriteLine("-----");
+
+            void Check(IterationMode mode, IterationStage stage)
+            {
+                int count = measurements.Count(m => m.Is(mode, stage));
+                Output.WriteLine($"Count({mode}{stage}) = {count}");
+                Assert.True(count > 0, $"AllMeasurements don't contain {mode}{stage}");
```
[the remark this answers is not in the document]
Comment (AndreyAkinshin) [reply tests/BenchmarkDotNet.IntegrationTests/EngineTests.cs:60]: Even with the given config, we still perform multiple iteration to evaluate the overhead.

[the remark this answers is not in the document]
Comment (adamsitnik) [reply tests/BenchmarkDotNet.IntegrationTests/EngineTests.cs:60]: > Even with the given config, we still perform multiple iteration to evaluate the overhead.

I see, thanks!

**Comment by AndreyAkinshin** [comment]
@adamsitnik thanks for the review, I have addressed all the comments.

**Pull request merged**